src/kernels/jupyter/launcher/jupyterExecution.ts

```diff
@@ -190,6 +193,32 @@ export class JupyterExecutionBase implements IJupyterExecution {
         return Promise.resolve(undefined);
     }
 
+    public async validateRemoteUri(uri: string): Promise<void> {
+        let connection: IJupyterConnection | undefined = undefined;
+        let sessionManager: IJupyterSessionManager | undefined = undefined;
+        try {
+            // Prepare our map of server URIs (needed in order to retrieve the uri during the connection)
+            await this.updateServerUri(uri);
+
+            // Create an active connection.
+            connection = await createRemoteConnectionInfo(uri, this.getServerUri.bind(this));
+
+            // Attempt to list the running kernels. It will return empty if there are none, but will
+            // throw if can't connect.
+            sessionManager = await this.jupyterSessionManagerFactory.create(connection, false);
```

Comment:
Password connect stuff should happen here too. This means we'll ask for user name password right after the user enters their URI

Reply:
perfect. was planning on adding that.